src/Workspaces/CSharp/Portable/Extensions/ITypeSymbolExtensions.TypeSyntaxGeneratorVisitor.cs

```diff
@@ -121,8 +122,8 @@ public TypeSyntax CreateSimpleTypeSyntax(INamedTypeSymbol symbol)
                 }
 
                 var typeArguments = symbol.IsUnboundGenericType
-                    ? Enumerable.Repeat(SyntaxFactory.OmittedTypeArgument(), symbol.TypeArguments.Length)
-                    : symbol.TypeArguments.Select(t => t.GenerateTypeSyntax());
+                    ? Enumerable.Repeat((TypeSyntax)SyntaxFactory.OmittedTypeArgument(), symbol.TypeArguments.Length)
+                    : symbol.TypeArguments.ZipAsArray(symbol.TypeArgumentsNullableAnnotations, (t, n) => t.WithNullability(n).GenerateTypeSyntax());
```

Comment:
oh ick :-/   I really hope your work is successful.  Because the actual symbol API here is pretty unpleasant.